include/fiction/algorithms/physical_design/post_layout_optimization.hpp

```diff
@@ -339,6 +339,71 @@ void optimize_output_positions(Lyt& lyt) noexcept
         }
     }
 }
+
+/**
+ * Utility function that checks and optimizes PO positions after each gate relocation iteration.
+ * This function moves POs that are not optimally positioned (e.g., in second rightmost or second bottom positions)
+ * to the optimal border positions by inserting buffer gates where the POs were.
+ *
+ * @tparam Lyt Cartesian gate-level layout type.
+ * @param lyt Gate-level layout.
+ * @param moved_gates Moved gates counter to decrement if PO is moved.
+ */
+template <typename Lyt>
+void check_and_optimize_po_positions(Lyt& lyt, uint64_t& moved_gates) noexcept
+{
+    // check all POs are in right (x=lyt.x()) or bottom (y=lyt.y()) border
+    lyt.foreach_po(
+        [&lyt, &moved_gates](const auto& po) noexcept
+        {
+            if (const auto tile = lyt.get_tile(lyt.get_node(po));
+                !(lyt.is_at_eastern_border(tile) || lyt.is_at_southern_border(tile)))
+            {
+                if ((tile.x == lyt.x() - 1) && (lyt.is_empty_tile({lyt.x(), tile.y})))
+                {
+                    // get fanin signal of the PO
+                    std::vector<mockturtle::signal<Lyt>> signals{};
+                    signals.reserve(1);
```

Comment:
`.reserve(1)` is most likely superfluous if you plan to store only a single element anyway.